### PR TITLE
Add Php 7 support

### PR DESCRIPTION
**Pull request by chrisryan**
Update to allow Php 7.
Update to new MongoDB driver.
Remove older versions of Php 5.4 and 5.5 from list.

### CODE REVIEW

File: tests/QueueTest.php [hunk 1]
```diff
@@ -172,10 +172,10 @@ public function ensureCountIndexWithPrefixOfPrevious()
         $this->queue->ensureCountIndex(['type' => 1, 'boo' => -1], false);
         $this->queue->ensureCountIndex(['type' => 1], false);
 
-        $this->assertSame(2, count($this->collection->getIndexInfo()));
+        $this->assertSame(2, count($this->collection->listIndexes()));
```
Comment (chadicus) [inline tests/QueueTest.php]: I'm not sure how it was working before, but the count of indexes should be 1. I think that's the issue w/ all of the ensure indexes tests

Comment (chrisryan) [inline tests/QueueTest.php]: I was able to figure it out after I trying it with 1.

**Comment by chrisryan** [comment]
I need to clean up all my commits but would grateful if anyone could tell me why scrutinizer does not recognize \MongoDB\BSON\ObjectID as a valid object.

**Comment by nubs** [comment]
I would assume it has something to do with that class coming from ext-mongo which is no longer in your composer dependencies.

**Comment by chrisryan** [comment]
That class is from the new ext-mongodb extension. I thought about putting the specific require in composer but wanted to avoid that as the mongodb/mongodb php package has the requires.

**Comment by chadicus** [comment]
@chrisryan The actual class name is `MongoDB\BSON\ObjectID`  note the uppercase `ID` i'm not sure if that makes a difference or not.

**Comment by chadicus** [comment]
@chrisryan the build is passing, if we can't figure out the scrutinizer stuff, then we can just merge and deal with it later

**Comment by chrisryan** [comment]
@chadicus fixed typo and pushed changes. I agree, we should move forward for now.

File: src/Queue.php [hunk 2]
```diff
@@ -429,20 +429,23 @@ private function ensureIndex(array $index)
...
                         return;
                     }
                 }
+
+                //@codeCoverageIgnoreStart
             }
         }
 
         throw new \Exception('couldnt create index after 5 attempts');
+        //@codeCoverageIgnoreEnd
```
Comment (chadicus) [inline src/Queue.php]: You could write a test that covers this if you passed in a collection mock to the constructor that threw a `\MongoDB\Exception\Exception` each time `createIndex` was called.

**Pull request merged**